src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -1854,60 +1868,73 @@ class RideConstructionWindow final : public Window
-                widgets[WIDX_BANK_STRAIGHT].tooltip = STR_RIDE_CONSTRUCTION_BRAKE_SPEED_LIMIT_TIP;
-                widgets[WIDX_BANK_RIGHT].tooltip = STR_RIDE_CONSTRUCTION_BRAKE_SPEED_LIMIT_TIP;
+                widgets[WIDX_SPEED_GROUPBOX].text = STR_RIDE_CONSTRUCTION_BRAKE_SPEED;
+                widgets[WIDX_SPEED_SETTING_SPINNER].tooltip = STR_RIDE_CONSTRUCTION_BRAKE_SPEED_LIMIT_TIP;
+                widgets[WIDX_SPEED_SETTING_SPINNER_UP].tooltip = STR_RIDE_CONSTRUCTION_BRAKE_SPEED_LIMIT_TIP;
+                widgets[WIDX_SPEED_SETTING_SPINNER_DOWN].tooltip = STR_RIDE_CONSTRUCTION_BRAKE_SPEED_LIMIT_TIP;
             }
             else
             {
-                widgets[WIDX_BANKING_GROUPBOX].text = STR_RIDE_CONSTRUCTION_BOOSTER_SPEED;
-                widgets[WIDX_BANK_LEFT].tooltip = STR_RIDE_CONSTRUCTION_BOOSTER_SPEED_LIMIT_TIP;
-                widgets[WIDX_BANK_STRAIGHT].tooltip = STR_RIDE_CONSTRUCTION_BOOSTER_SPEED_LIMIT_TIP;
-                widgets[WIDX_BANK_RIGHT].tooltip = STR_RIDE_CONSTRUCTION_BOOSTER_SPEED_LIMIT_TIP;
+                widgets[WIDX_SPEED_GROUPBOX].text = STR_RIDE_CONSTRUCTION_BOOSTER_SPEED;
+                widgets[WIDX_SPEED_SETTING_SPINNER].tooltip = STR_RIDE_CONSTRUCTION_BOOSTER_SPEED_LIMIT_TIP;
+                widgets[WIDX_SPEED_SETTING_SPINNER_UP].tooltip = STR_RIDE_CONSTRUCTION_BOOSTER_SPEED_LIMIT_TIP;
+                widgets[WIDX_SPEED_SETTING_SPINNER_DOWN].tooltip = STR_RIDE_CONSTRUCTION_BOOSTER_SPEED_LIMIT_TIP;
             }
 
             _currentlyShowingBrakeOrBoosterSpeed = true;
-            widgets[WIDX_BANK_LEFT].text = STR_RIDE_CONSTRUCTION_BRAKE_SPEED_VELOCITY;
-
-            widgets[WIDX_BANK_LEFT].type = WindowWidgetType::Spinner;
-            widgets[WIDX_BANK_LEFT].left = 12;
-            widgets[WIDX_BANK_LEFT].right = 96;
-            widgets[WIDX_BANK_LEFT].top = 138;
-            widgets[WIDX_BANK_LEFT].bottom = 149;
-            widgets[WIDX_BANK_STRAIGHT].type = WindowWidgetType::Button;
-            widgets[WIDX_BANK_STRAIGHT].text = STR_NUMERIC_UP;
-            widgets[WIDX_BANK_STRAIGHT].left = 84;
-            widgets[WIDX_BANK_STRAIGHT].right = 95;
-            widgets[WIDX_BANK_STRAIGHT].top = 139;
-            widgets[WIDX_BANK_STRAIGHT].bottom = 148;
-            widgets[WIDX_BANK_RIGHT].type = WindowWidgetType::Button;
-            widgets[WIDX_BANK_RIGHT].text = STR_NUMERIC_DOWN;
-            widgets[WIDX_BANK_RIGHT].left = 72;
-            widgets[WIDX_BANK_RIGHT].right = 83;
-            widgets[WIDX_BANK_RIGHT].top = 139;
-            widgets[WIDX_BANK_RIGHT].bottom = 148;
-            hold_down_widgets |= (1uLL << WIDX_BANK_STRAIGHT) | (1uLL << WIDX_BANK_RIGHT);
-        }
-
-        widgets[WIDX_BANKING_GROUPBOX].right = 162;
+            widgets[WIDX_SPEED_SETTING_SPINNER].text = STR_RIDE_CONSTRUCTION_BRAKE_SPEED_VELOCITY;
+
+            widgets[WIDX_SPEED_SETTING_SPINNER].type = WindowWidgetType::Spinner;
```

Comment:
Why are all these changes needed?

Reply:
If you are asking why the UI was changed to accomodate speed and seat rotation at the same time, it's because block brakes support both.